python/adbc_driver_manager/meson.build

```diff
@@ -0,0 +1,51 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+project(
+    'adbc_driver_manager',
+    'cython',
+    version: run_command(
+        ['python', 'adbc_driver_manager/_version.py'],
+        check: true,
+    ).stdout().replace(
+        'Version:',
+        '',
+    ).strip(),
+    license: 'Apache-2.0',
+    meson_version: '>=1.2.1',
+    default_options: ['buildtype=release', 'warning_level=2', 'cpp_std=c++17'],
+)
+
+backward_dep = dependency('backward-cpp')
+# hack until https://github.com/mesonbuild/wrapdb/pull/2072
+backward_incdir = include_directories('subprojects/backward-cpp-1.6')
+
+adbc_driver_manager_dep = dependency(
```

Comment:
The idea here is that Meson can look for `adbc-driver-manager` through its typical its typical search mechanism (typically pkgconf). If it is not installed, Meson will look at `subprojects/arrow-adbc`, which symlinks to the C directory

One caveat to this is if a user is trying to locally develop on a system where the ADBC Driver Manager is installed, it will pick up the system dependency rather than the local one. If we wanted to prevent that we could add more logic potentially